app/gui2/src/stores/visualization.ts

```diff
@@ -0,0 +1,285 @@
+import { fileName } from '@/util/file'
+import Compiler from '@/workers/visualizationCompiler?worker'
+import * as vue from 'vue'
+import { type DefineComponent } from 'vue'
+import * as vueUseCore from '@vueuse/core'
+import VisualizationContainer from '@/components/VisualizationContainer.vue'
+import * as useVisualizationConfig from '@/providers/useVisualizationConfig'
+
+import { defineStore } from 'pinia'
+
+const moduleCache: Record<string, any> = {
+  vue,
+  '@vueuse/core': vueUseCore,
+  'builtins/VisualizationContainer.vue': { default: VisualizationContainer },
+  'builtins/useVisualizationConfig.ts': useVisualizationConfig,
+}
+// @ts-expect-error Intentionally not defined in `env.d.ts` as it is a mistake to access anywhere
+// else.
+window.__visualizationModules = moduleCache
+
+export type Visualization = DefineComponent<{ data: {} | string }>
+type VisualizationModule = {
+  default: Visualization
+  name?: string
+  inputType?: string
+  scripts?: string[]
+  styles?: string[]
+}
+
+const builtinVisualizationPaths: Record<string, string> = {
+  JSON: '/visualizations/JSONVisualization.vue',
+  Table: '/visualizations/TableVisualization.vue',
+  Scatterplot: '/visualizations/ScatterplotVisualization.vue',
+  Histogram: '/visualizations/HistogramVisualization.vue',
+  Heatmap: '/visualizations/HeatmapVisualization.vue',
+  'SQL Query': '/visualizations/SQLVisualization.vue',
+  'Geo Map': '/visualizations/GeoMapVisualization.vue',
+  Image: '/visualizations/ImageBase64Visualization.vue',
+  Warnings: '/visualizations/WarningsVisualization.vue',
```

Comment:
I'm not super sure if we want to keep bulitin visualizations in `public` folder. It works for now, but I consider it a bit hacky. At minimum, we should be able to still build and minify them in production mode to reduce their footprint on application startup time.

This is mostly for the future, I don't think this needs to be changed in this PR. But you could leave a comment to the code.

Reply:
i *think* it should be simple enough to change it